### ISSUE TITLE
Feature for native DB based RAG 

### ISSUE THREAD

**Issue by Anindyadeep** [premAI-io/text2sql#13]
Full end to end support for DB based RAG where user can choose

1. retriever 
2. The LLM 
3. The embedding model
4. The algorithm 
etc and they will get an end to end solution for DB based rag. For just the starter, start with Qdrant as the vector DB and sqlite as the DB.

**Comment by ansarizafar** [comment]
We can also use LibSQL which is a fork of SQLite with built-in support for vector search https://turso.tech/blog/turso-brings-native-vector-search-to-sqlite?webui=true&type=signup&content=ai

**Comment by Anindyadeep** [comment]
yes great point, even I also have been looking into this project for some time. We will be adding a whole new class of retrievers in the upcoming releases. Thanks for sharing

**Comment by ansarizafar** [comment]
KuzuDB https://kuzudb.com/ is another good option much better than Sqlite.
